### PR TITLE
Miri

### PR DESCRIPTION
**Pull request by Eli-BB8**
Integration with Miri data source  ⚙️

### CODE REVIEW

File: src/util/matchToKartoffel.js [hunk 1]
```diff
@@ -343,6 +345,155 @@ const match_nv_sql = (obj, dataSource) => {
...
+                delete obj[rawKey];
+                break;
+            //hierarchy
+            case fn[dataSource].hierarchy:
+                let hr = obj[rawKey].replace('\\', '/');
+                if (hr.includes('/')) {
+                    hr = hr.split('/').map(unit => unit.trim());
+
+                    for(const [index, value] of hr.entries()) {
+                        if (isStrContains(value, [`${obj[fn[dataSource].firstName]} ${obj[fn[dataSource].lastName]}`, '-', ''])) {
```
Comment (weizmm) [inline src/util/matchToKartoffel.js]: what about ''?
`"kujhkh".includes('')
true`

File: src/util/matchToKartoffel.js [hunk 2]
```diff
@@ -343,6 +345,155 @@ const match_nv_sql = (obj, dataSource) => {
...
+                            hr.splice(index);
+                            break;
+                        }
+                    }
+
+                    hr = hr.join('/');
+                }
+
+                obj.hierarchy = `${fn.rootHierarchy.city}${company ? '/' + company : ''}${hr ? '/' + hr : ''}`;
```
Comment (weizmm) [inline src/util/matchToKartoffel.js]: use with the variable above 
:arrow_double_up:

File: src/util/matchToKartoffel.js [hunk 3]
```diff
@@ -343,6 +345,155 @@ const match_nv_sql = (obj, dataSource) => {
...
+                        uniqueID: obj[fn[dataSource].domainUsers],
+                        dataSource
+                      }
+                    ];
+                }
+                // set identityCard || personlNumber if needed
+                if (!obj.hasOwnProperty('identityCard') ||
+                    !obj.hasOwnProperty('personalNumber') ||
+                    !obj.hasOwnProperty(fn[dataSource].identityCard) ||
+                    !obj.hasOwnProperty(fn[dataSource].identityCard)) {
```
Comment (weizmm) [inline src/util/matchToKartoffel.js]: change to personalNumber

File: src/config/fieldNames.js [hunk 4]
```diff
@@ -1,128 +1,159 @@
...
+    nvSQL_name: {
+        firstName: "KfirstName",
+        lastName: "KlastName",
+        uniqueID: "unid",
+        hierarchy: "hierarchy",
+        pn: "pn",
+        identityCard: "id",
+        uniqeFieldForDeepDiff: "unid",
+    },
+    city_name: {
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply src/config/fieldNames.js]: ✔️

File: mocks/mockGen.js [hunk 5]
```diff
@@ -116,7 +122,44 @@ for (let i = 0; i < esAmount; i++) {
...
+                                   null,
+                                   null,
+                                   "",
+                                   "",
+                                   faker.internet.email().split('@')[0] + '@' + miriTypes.miriMail]);
+    miriUser.tz = utils.randomElement([utils.generateID(), '', 'לא ידוע', null]);
+    miriUser.personalNumber = utils.randomElement([mis[miriAkaStart + i], '', 'לא ידוע', null]);
+    miriUser.rank = utils.randomElement(dataTypes.RANK);
+    miriUser.rld = utils.randomElement([faker.date.between(faker.date.future(10), faker.date.past(10)).toISOString().split('T')[0] + " 00:00:00.0"], null, "", 'לא ידוע');
+    miriUser.job = faker.name.jobTitle();
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply mocks/mockGen.js]: 🆗

File: src/config/fieldNames.js [hunk 6]
```diff
@@ -1,128 +1,159 @@
...
+        profession: 'profession',
+        hierarchy: 'hr',
+        domainUsers: 'domUser',
+        compnay: 'company',
+        entityTypePrefix: {
+            s: ['e', 'l', 'i'],
+            c: ['g', 'j'],
+            gu: ['m'],
+        },
+        company: "compnay",
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply src/config/fieldNames.js]: ❌

File: src/util/completeFromAka.js [hunk 7]
```diff
@@ -105,6 +127,10 @@ module.exports = (obj, akaData, dataSource) => {
                     if (obj.entityType === fn.entityTypeValue.c) { break; };
                     complete_adNN(obj, akaRecord);
                     break;
+                case fn.dataSources.city:
+                    if (obj.entityType === fn.entityTypeValue.c) { break; };
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply src/util/completeFromAka.js]: ▶️ ◀️

File: src/util/fieldsUtils/domainUserHandler.js [hunk 8]
```diff
@@ -10,11 +10,19 @@ const Auth = require('../../auth/auth');
  * @param {*} person The Person object that returned from kartoffel
  * @param {*} record The raw object that coming from the dataSource
  * @param {*} dataSource The dataSource of the object
- * 
+ *
  *  */
 module.exports = async (person, record, dataSource) => {
+    let dataSourceName = dataSource;
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply src/util/fieldsUtils/domainUserHandler.js]: ❌

File: src/util/generalUtils/strignContains.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+module.exports = (target, pattern) => {
+    let value = 0;
+
+    pattern.forEach(function(word){
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply src/util/generalUtils/strignContains.js]: ▶️

File: src/util/matchToKartoffel.js [hunk 10]
```diff
@@ -343,6 +345,155 @@ const match_nv_sql = (obj, dataSource) => {
     })
 };
 
+const match_city = (obj, dataSource) => {
+    const objKeys = Object.keys(obj);
+    // initialize variables for hierarchy matching and define default hierarchy
+    obj.hierarchy = `${fn.rootHierarchy.city}${obj.company ? '/' + obj.company : ''}`;
```
[the remark this answers is not in the document]
Comment (Eli-BB8) [reply src/util/matchToKartoffel.js]: 👍

File: src/util/matchToKartoffel.js (hunk 1 above)
Comment (Eli-BB8) [reply src/util/matchToKartoffel.js]: 👍

File: src/util/matchToKartoffel.js (hunk 2 above)
Comment (Eli-BB8) [reply src/util/matchToKartoffel.js]: 👍

File: src/util/matchToKartoffel.js (hunk 3 above)
Comment (Eli-BB8) [reply src/util/matchToKartoffel.js]: 💯

**Pull request merged**